pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -42,12 +43,62 @@ type nodeConfig struct {
 	node *v1.Node
 }
 
+// discoverKubeAPIServerEndpoint discovers the kubernetes api server endpoint from the
+func discoverKubeAPIServerEndpoint() (string, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+	if len(cfg.Host) == 0 {
+		return "", errors.Wrap(err, "could not get host name for the kubernetes api server")
+	}
+	return cfg.Host, nil
+}
+
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
-func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) *nodeConfig {
-	return &nodeConfig{k8sclientset: clientset, Windows: windows.New(windowsVM)}
+func NewNodeConfig(clientset *kubernetes.Clientset, windowsVM types.WindowsVM) (*nodeConfig, error) {
+	var workerIgnitionEndpoint string
+	var err error
+	if nodeConfigCache.workerIgnitionEndPoint == "" {
+		var kubeAPIServerEndpoint string
+		// We couldn't find it in cache. Let's compute it now.
+		kubeAPIServerEndpoint, err = discoverKubeAPIServerEndpoint()
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to find kube api server endpoint")
+		}
+		clusterAddress, err := getClusterAddr(kubeAPIServerEndpoint)
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting cluster address")
+		}
+		workerIgnitionEndpoint = "https://api-int." + clusterAddress + ":22623/config/worker"
+		nodeConfigCache.workerIgnitionEndPoint = workerIgnitionEndpoint
+	} else {
+		workerIgnitionEndpoint = nodeConfigCache.workerIgnitionEndPoint
+	}
```

Comment:
This `else` block and the local variable `workerIgnitionEndpoint` is not needed if we just use `nodeConfigCache.workerIgnitionEndPoint`  directly in `&nodeConfig`.